features/upgrade/sdn/policy-upgrade.feature

```diff
@@ -78,3 +78,116 @@ Feature: SDN compoment upgrade testing
       | curl | -s | --connect-timeout | 5 | <%= cb.pod2ip %>:8080 |
     Then the step should fail
     And the output should not contain "Hello"
+
+
+  # @author asood@redhat.com
+  @admin
+  @upgrade-prepare
+  Scenario: Check the namespace networkpolicy for an application works well after upgrade - prepare
+    Given I switch to cluster admin pseudo user
+    When I run the :new_project client command with:
+      | project_name | test1 |
+    Then the step should succeed
+    When I run the :new_project client command with:
+      | project_name | test2 |
+    Then the step should succeed
+
+    When I use the "test1" project
+    Given I obtain test data file "networking/list_for_pods.json"
+    And I run the :create client command with:
+      | f | list_for_pods.json |
+    Then the step should succeed
+    Given 2 pods become ready with labels:
+      | name=test-pods |
+    And evaluation of `pod(1).ip` is stored in the :pod2ip clipboard
+    And evaluation of `pod(0).name` is stored in the :pod1 clipboard
+
+    Given I obtain test data file "rc/idle-rc-1.yaml"
+    And I run the :create client command with:
+      | f | idle-rc-1.yaml |
+    Then the step should succeed
+    Given 2 pods become ready with labels:
+      | name=hello-idle |
+    And evaluation of `pod(2).name` is stored in the :pod3 clipboard
+
+    When I use the "test2" project
+    Given I obtain test data file "networking/list_for_pods.json"
+    And I run the :create client command with:
+      | f | list_for_pods.json |
+    Then the step should succeed
+    Given 2 pods become ready with labels:
+      | name=test-pods |
+    And evaluation of `pod(4).name` is stored in the :pod5 clipboard
+
+    Given I obtain test data file "rc/idle-rc-1.yaml"
+    And I run the :create client command with:
+      | f | idle-rc-1.yaml |
+    Then the step should succeed
+    Given 2 pods become ready with labels:
+      | name=hello-idle |
+    And evaluation of `pod(6).name` is stored in the :pod7 clipboard
+
+    Given I have a project
+    When I run the :label admin command with:
+      | resource | namespace                |
+      | name     | test2                    |
+      | key_val  | team=operations          |
+    Then the step should succeed
+
+    Given the AllowNamespaceAndPod policy is applied to the "test1" namespace
+    Then the step should succeed
+
+    When I use the "test1" project
+    When I execute on the "<%= cb.pod1 %>" pod:
+      | curl | -s | --connect-timeout | 5 | <%= cb.pod2ip %>:8080 |
+    Then the step should fail 
+    And the output should not contain "Hello"
+    When I execute on the "<%= cb.pod3 %>" pod:
+      | curl | -s | --connect-timeout | 5 | <%= cb.pod2ip %>:8080 |
+    Then the step should fail
+    And the output should not contain "Hello"
+
+    When I use the "test2" project
+    When I execute on the "<%= cb.pod5 %>" pod:
+      | curl | -s | --connect-timeout | 5 | <%= cb.pod2ip %>:8080 |
+    Then the step should succeed
```

Comment:
Here we need to add one step to make sure 'pod5 cannot access pod3.ip ( since pod do not have correct label)

Reply:
pod3 has label name=hello-idle, pod5 has label name=test-pods.
So pod with label name=test-pods not being accessed by pod with label name=hello-idle is covered by the following steps:-

When I execute on the "<%= cb.pod7 %>" pod:
      | curl | -s | --connect-timeout | 5 | <%= cb.pod2ip %>:8080 |
    Then the step should fail
    And the output should not contain "Hello"
Though the pods are in different namespaces/project where pod2ip is ip of the pod with label name=test-pods and pod7 is pod with label name=hello-idle, these steps do test the flow you are referring to.